src/View/Escape/EscapeAbstract.php

```diff
@@ -0,0 +1,80 @@
+<?php
+declare(strict_types=1);
+
+namespace YapepBase\View\Escape;
+
+use YapepBase\Exception\ParameterException;
+
+/**
+ * Common ancestor for the different escape classes.
+ */
+abstract class EscapeAbstract implements IEscape
+{
+    protected $objectToEscape;
+
+    abstract protected function escapeString(string $value);
+
+    abstract protected function escapeArray(array $array);
+
+    abstract protected function escapeBool(bool $value);
+
+    abstract protected function escapeInt(int $value);
+
+    abstract protected function escapeFloat(float $value);
+
+    abstract protected function escapeNull();
+
+    public function __escape($value)
+    {
+        switch (gettype($value)) {
+            case 'boolean':
+                return $this->escapeBool($value);
+                break;
+
+            case 'integer':
+                return $this->escapeInt($value);
+                break;
+
+            case 'double':
+                return $this->escapeFloat($value);
+                break;
+
+            case 'string':
+                return $this->escapeString($value);
+                break;
+
+            case 'array':
+                return $this->escapeArray($value);
+                break;
+
+            case 'object':
+                return $this->wrapObjectForEscape($value);
+                break;
+
+            case 'NULL':
+                return $this->escapeNull();
+                break;
+
+            default:
+                return null;
+                break;
+        }
+    }
+
+    protected function wrapObjectForEscape(object $value)
+    {
+        foreach ($value as $whatever) {
```

Comment:
Beautiful...

Reply:
I could have called it $thisWillBeIgnoredForeachIsJustNeededBecausePhpCanBeQuiteProblematicSomeTimes but $whatever describes it well enough and much shorter